packages/yew-bootstrap/src/component/form/form_control.rs

```diff
@@ -82,11 +82,7 @@ pub struct FormControlProps {
 /// Convert an option (Typically integer) to an AttrValue option
 fn convert_to_string_option<T>(value: &Option<T>) -> Option<AttrValue>
 where T: std::fmt::Display {
-    if let Some(value) = value {
-        Some(AttrValue::from(value.to_string()))
-    } else {
-        None
-    }
+    value.as_ref().map(|v| AttrValue::from(v.to_string()))
```

Comment:
Ok thanks for this type of change, it's quite useful. However I don't understand how it gets an Option and returns an option, is there something special with "as_ref"?

Reply:
My most honest answer is that I let `cargo clippy` guide me towards using `as_ref()`. Try running it in one of your projects and it'll give very helpful advice. I have it integrated into my IDE.

If I try to understand why this is necessary, first I'd go [here](https://doc.rust-lang.org/std/option/) and read about `Option`s. Here it says:

> [as_ref](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref) converts from [&](https://doc.rust-lang.org/std/primitive.reference.html)[Option](https://doc.rust-lang.org/std/option/enum.Option.html)<T> to [Option](https://doc.rust-lang.org/std/option/enum.Option.html)<[&](https://doc.rust-lang.org/std/primitive.reference.html)T>

If you do this:

```rust
value.map(|v| AttrValue::from(v.to_string()))
```

Then the closure is taking ownership of v - it moves it from its source. The borrow checker stops us from doing this, I think because we are getting an immutable reference to the option, so we are not allowed to change its contents directly.

If instead we use `as_ref()`, then "v" is a reference, it is not moved into the closure. `v.to_string()` creates a new object. `map` then operates on the immutable reference to the Option with an immutable reference inside of it and creates a new `Option` with an "owned" String inside of it. 

My understanding above may be a bit incorrect, I am not yet an expert. `rustc` and `cargo clippy` will mostly lead you to the right answers, even if you don't understand them at first.

Reply:
I asked for feedback about my above comments in the Rust Programming Language Community Server (Discord) and there were two points raised by the folks there:
> the answer to the question is that you used map. as_ref doesn't have any bearing

and 

> &Option<T> is always a pain to deal with
> Definitely always prefer Option<&T> if at all possible
> It's easier to work with the &T, otherwise you have to hope that T is `Clone` which it might not be. 

I might poke around with this function and maybe try to eliminate it entirely if possible.